linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker
+
+import scala.concurrent._
+
+import java.nio.charset.StandardCharsets.UTF_8
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.ir.ClassKind
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.logging._
+
+import org.scalajs.junit.async._
+
+import org.scalajs.linker._
+import org.scalajs.linker.interface._
+
+import org.scalajs.linker.testutils._
+import org.scalajs.linker.testutils.TestIRBuilder._
+
+class IncrementalTest {
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  import IncrementalTest._
+
+  @Test // #4335
+  def testChangeMethodReachedFromExport(): AsyncResult = await {
```

Comment:
I tried to follow #4266 as good as I could for this, but it seems the location of `@Test` and how to reference issues (in the test name or comment) is not always the same.